tests/queries/tests.py

```diff
@@ -1314,7 +1314,7 @@ def test_ticket7107(self):
     def test_ticket8683(self):
         # An error should be raised when QuerySet.datetimes() is passed the
         # wrong type of field.
-        with self.assertRaisesMessage(AssertionError, "'name' isn't a DateTimeField."):
+        with self.assertRaisesMessage(AssertionError, "'name' isn't a DateField or DateTimeField."):
```

Comment:
Other than this specific line, I believe all concerns have been addressed. This particular change was as a result of removing the specific DateTime and Date expressions, and switching .dates() and .datetimes() to using the generic `Trunc` class.